### DISCUSSION TITLE
Torguard + Wireguard port forwarding

### DISCUSSION THREAD

**Discussion by perry-mitchell** [qdm12/gluetun#1282] [Q&A]
Trying to get port forwarding working via a custom configuration using Wireguard with Torguard. I've followed the wiki instructions, and I can connect to the specified Torguard server IP. I have also confirmed that the container's traffic is routed correctly via Torguard. The problem is that I can't seem to figure out why the ports aren't open or why I can't access services running on my server via the tunnel.

I've removed my old port forwards that were working with OpenVPN, and have changed them all for Wireguard ones, on Torguard:

![image](https://user-images.githubusercontent.com/3869469/206868900-79fe7538-1f2b-4e33-9865-618163da81e8.png)

The relevant docker launch command:

```
docker run -d --name='GluetunCustom' --net='br0' --ip='192.168.0.177'
  -e TZ="Europe/Kiev" -e HOST_OS="Unraid" -e HOST_HOSTNAME="<snip>"
  -e HOST_CONTAINERNAME="GluetunCustom" -e 'TIMEZONE'='<snip>'
  -e 'VPN_SERVICE_PROVIDER'='custom' -e 'VPN_TYPE'='wireguard'
  -e 'VPN_ENDPOINT_IP'='<snip>' -e 'VPN_ENDPOINT_PORT'='1443'
  -e 'WIREGUARD_PUBLIC_KEY'='<snip>' -e 'WIREGUARD_PRIVATE_KEY'='<snip>'
  -e 'WIREGUARD_ADDRESSES'='10.<snip>/24'
  -e 'FIREWALL_VPN_INPUT_PORTS'='54321,51234' -e 'FIREWALL'='off'
  -e 'LOG_LEVEL'='info' -e 'FIREWALL_INPUT_PORTS'='8112,58846,54321,51234'
  -e 'FIREWALL_DEBUG'='on' -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8000]'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/qdm12/gluetun/master/doc/logo_256.png'
  -v '/mnt/user/appdata/gluetun2':'/gluetun':'rw' --cap-add=NET_ADMIN
  --restart always 
  'qmcgaw/gluetun:latest'
```

I'm using `ncat -l 54321` etc. to test access, from within the container. Curling `http://<public_ip>:54321` yields errors as the port isn't open.

**Comment by perry-mitchell** [comment]
Got the following from Torguard support:

> Please note wireguard containers will not work with port forward, this is a limitation from virtualization, if you try it on main host it will work ok.

**Reply by perry-mitchell** [reply]
And then later, when I admitted my disbelief:

> It was tested multiple times with docker, with other Virtualizations it works, but not docker, it is not about our support really, wireguard port forward works on all OS, it is just docker implementation for torrent container that seems not to listen correctly on wireguard.

**Reply by qdm12** [reply]
It most likely doesn't have anything to do with Docker. Vpn server side port forwarding works fine on private internet access, VyprVPN, protonvpn and worked fine on Mullvad too.

Have you tried it outside Docker, does it work? My wild guess would be that port forwarding doesn't work with wireguard because of a vpn server implementation issue
